Lib/queue.py

```diff
@@ -182,6 +182,15 @@ def get(self, block=True, timeout=None):
             self.not_full.notify()
             return item
 
+    def front(self):
+        ''' Get the first element in the queue without removing
+ 
+        If queue is empty, then return None
+        '''
+        if not self.empty():
+            return self.queue[0]
```

Comment:
This has a race condition: a different thread could pop something off of the queue between the execution of these two lines.

Reply:
Should we apply condition variables like this?

```  
    def front(self):
        ''' Get the first element in the queue without removing
 
        If queue is empty, then return None
        '''
        with self.all_tasks_done:
            while self.unfinished_tasks:
                self.all_tasks_done.wait()
                
                if self._qsize():
                    return self.queue[0]
                return None
```

Reply:
You may be able to just do `return self.queue[0]` with `try: ... except IndexError:`. But as I mentioned in the issue, I'm not sure we should be adding such a method at all.

Note also that this has unexpected results for `LifoQueue`, which inherits this method from `Queue`.